selfdrive/car/gm/interface.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+import time
+import numpy as np
+
+from selfdrive.config import Conversions as CV
+from selfdrive.car.gm.carstate import CarState
+from selfdrive.car.gm.carcontroller import CarController
+from selfdrive.boardd.boardd import can_capnp_to_can_list
+
+from cereal import car
+
+import zmq
+from selfdrive.services import service_list
+import selfdrive.messaging as messaging
+
+# Car button codes
+class CruiseButtons:
+  UNPRESS     = 2
+  RES_ACCEL   = 4
+  DECEL_SET   = 6
+  CANCEL      = 12
+  MAIN        = 10
+
+# Car chimes, beeps, blinker sounds etc
+class CM:
+  TOCK = 0x81
+  TICK = 0x82
+  LOW_BEEP = 0x84
+  HIGH_BEEP = 0x85
+  LOW_CHIME = 0x86
+  HIGH_CHIME = 0x87
+
+class CarInterface(object):
+  def __init__(self, CP, logcan, sendcan=None):
+    self.logcan = logcan
+    self.CP = CP
+
+    self.frame = 0
+    self.can_invalid_count = 0
+
+    # *** init the major players ***
+    self.CS = CarState(CP, self.logcan)
+
+    # sending if read only is False
+    if sendcan is not None:
+      self.sendcan = sendcan
+      self.CC = CarController()
+
+  # returns a car.CarState
+  def update(self):
+    # ******************* do can recv *******************
+    can_pub_main = []
+    canMonoTimes = []
+    for a in messaging.drain_sock(self.logcan):
+      canMonoTimes.append(a.logMonoTime)
+      # Not listening on object detection (0 or 0x11 from Panda)
+      can_pub_main.extend(can_capnp_to_can_list(a.can, [1, 0x14]))
+    self.CS.update(can_pub_main)
+
+    # create message
+    ret = car.CarState.new_message()
+
+    # speeds
+    ret.vEgo = self.CS.v_ego
+    ret.wheelSpeeds.fl = self.CS.cp.vl[0x106b8040]['WHEEL_SPEED_FL']
+    ret.wheelSpeeds.fr = self.CS.cp.vl[0x106b8040]['WHEEL_SPEED_FR']
+    ret.wheelSpeeds.rl = self.CS.cp.vl[0x106b8040]['WHEEL_SPEED_RL']
+    ret.wheelSpeeds.rr = self.CS.cp.vl[0x106b8040]['WHEEL_SPEED_RR']
+
+    # gas pedal information.
+    ret.gas = self.CS.pedal_gas / 254.0
+    if not self.CP.enableGas:
+      ret.gasPressed = self.CS.pedal_gas > 0
+    else:
+      ret.gasPressed = self.CS.user_gas_pressed
+
+    # brake pedal
+    ret.brake = self.CS.user_brake
+    ret.brakePressed = self.CS.brake_pressed != 0
+
+    # steering wheel
+    ret.steeringAngle = self.CS.angle_steers
+    ret.steeringPressed = self.CS.steer_override
```

Comment:
This is always being set to False. We need the timer to reset when the user uses the steering wheel.